### PR TITLE
A set of 80 standard TrueType fonts from URW

### PR DESCRIPTION
**Pull request by mhavo**
The set of 80 standard Truetype fonts from URW and included in the
GhostPCL -download archives.

After making all changes to the cask:

- [ ] `brew cask audit --download {{cask_file}}` is error-free.
- [ ] `brew cask style --fix {{cask_file}}` reports no offenses.
- [ ] The commit message includes the cask’s name and version.

This package contains many different fonts. Seems to be very hard to get everything right.

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not already refused in [closed issues].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-fonts/pulls
[closed issues]: https://github.com/caskroom/homebrew-fonts/issues?q=is%3Aissue+is%3Aclosed
[the correct repo]: https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

File: Casks/font-urwfonts.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
+cask 'urwfonts-8.71' do
```
Comment (vitorgalvao) [inline Casks/font-urwfonts.rb]: This needs to be the same as the cask’s name (minus the `.rb`).

You **really** need to run the first three commands in the template. They catch errors like this.

File: Casks/font-urwfonts.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  version '1.05'
```
Comment (vitorgalvao) [inline Casks/font-urwfonts.rb]: `8.71`.

File: Casks/font-urwfonts.rb [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  sha256 '8e5eb990d6b21e700747591f99754145213753446bcfff2881e08da5eb8d0027'
+
+  url 'http://ftp.gwdg.de/pub/misc/ghostscript/AFPL/GhostPCL/urwfonts-8.71.tar.bz2'
```
Comment (vitorgalvao) [inline Casks/font-urwfonts.rb]: I’m not finding this specific download on the page.

Comment (commitay) [inline Casks/font-urwfonts.rb]: Please provide a link to where you got this URL from, I can't find it on the `homepage`

Comment (vitorgalvao) [reply Casks/font-urwfonts.rb]: Instead use `#{version}` and double quotes. As in `url "http://ftp.gwdg.de/pub/misc/ghostscript/AFPL/GhostPCL/urwfonts-#{version}.tar.bz2"`.

File: Casks/font-urwfonts.rb [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  name 'urwfonts-8.71'
```
Comment (vitorgalvao) [inline Casks/font-urwfonts.rb]: No versions in names. Also, isn’t there a better name?

File: Casks/font-urwfonts.rb (hunk 1 above)
Comment (vitorgalvao) [reply Casks/font-urwfonts.rb]: Does the name follow [the rules](https://github.com/caskroom/homebrew-fonts/blob/master/CONTRIBUTING.md#naming-font-casks)?

File: Casks/font-urwfonts.rb [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  homepage 'https://www.ghostscript.com/GhostPCL.html'
+
+  font 'urwfonts-8.71/A028-Ext.ttf'
```
Comment (vitorgalvao) [inline Casks/font-urwfonts.rb]: Instead use `#{version}` and double quotes. In this and all subsequent lines

File: Casks/font-urwfonts.rb [hunk 6]
```diff
@@ -0,0 +1,88 @@
+cask 'font-urwfonts' do
+  version '1.05'
+  sha256 '8e5eb990d6b21e700747591f99754145213753446bcfff2881e08da5eb8d0027'
+
+  url 'http://ftp.gwdg.de/pub/misc/ghostscript/AFPL/GhostPCL/urwfonts-8.71.tar.bz2'
```
Comment (vitorgalvao) [inline Casks/font-urwfonts.rb]: I’m not finding this specific download on the page.

Comment (vitorgalvao) [reply Casks/font-urwfonts.rb]: Instead use `#{version}` and double quotes. As in `url "http://ftp.gwdg.de/pub/misc/ghostscript/AFPL/GhostPCL/urwfonts-#{version}.tar.bz2"`.

File: Casks/font-urwfonts.rb [hunk 7]
```diff
@@ -0,0 +1,88 @@
+cask 'font-urwfonts' do
+  version '1.05'
```
Comment (vitorgalvao) [inline Casks/font-urwfonts.rb]: `8.71`

File: Casks/font-urwfonts.rb (hunk 6 above)
Comment (mhavo) [reply Casks/font-urwfonts.rb]: URW-fonts are normally included inside the GhostPCL -download archives. That is why I am using this ftp-mirror site and direct download link to the tar-ball.

File: Casks/font-urwfonts.rb (hunk 4 above)
Comment (mhavo) [reply Casks/font-urwfonts.rb]: No. There is just this name for this package.

File: Casks/font-urwfonts.rb (hunk 6 above)
Comment (vitorgalvao) [reply Casks/font-urwfonts.rb]: Unless we can verify this specific URL from the homepage, we cannot accept this cask.

File: Casks/font-urwfonts.rb [hunk 8]
```diff
@@ -0,0 +1,88 @@
+cask 'font-urwfonts' do
+  version '8.71'
+  sha256 '8e5eb990d6b21e700747591f99754145213753446bcfff2881e08da5eb8d0027'
+
+  url "http://ftp.gwdg.de/pub/misc/ghostscript/AFPL/GhostPCL/urwfonts-#{version}.tar.bz2"
+  name "urwfonts-#{version}"
```
Comment (vitorgalvao) [inline Casks/font-urwfonts.rb]: `#{version}` is to be used in the `font` stanzas.

There are no versions in `names`.

File: Casks/font-urwfonts.rb (hunk 4 above)
Comment (vitorgalvao) [reply Casks/font-urwfonts.rb]: This is supposed to be the name of the font, not the name of the package.

File: Casks/font-urwfonts.rb (hunk 8 above)
Comment (mhavo) [reply Casks/font-urwfonts.rb]: Sorry about that.

**Comment by vitorgalvao** [comment]
Please don’t keep making changes until you tell us where you got the download URL from. If we can’t verify its authenticity, we won’t merge the cask.

**Comment by mhavo** [comment]
Oh. Ok. It is public ftp-mirror. Is that bad? Look at the main site: http://ftp.gwdg.de/

**Comment by vitorgalvao** [comment]
There’s nothing on the homepage (https://www.ghostscript.com/GhostPCL.html) that indicates that FTP is officially supported.

**Comment by mhavo** [comment]
Ok. You can by all means disregard this pull request then. But let me explain more. Ghostscript is very, very old well known package and URW-fonts have been with that package a long time too. From the official site you can find only the whole package. I really don't know how to install fonts from inside two packages. That bz2 what you can find from this FTP is all over the internet and you can read the license of the fonts: they are free.

**Comment by mhavo** [comment]
Maybe we can use this CTAN-archive then, but from there you can have older version...

https://ctan.org/tex-archive/obsolete/support/ghostscript/AFPL/GhostPCL

**Comment by vitorgalvao** [comment]
@mhavo You are not understanding. We don’t care what the package is, we care *where it comes from*. Your download `url` is nowhere to be found on the `homepage`, **which means someone auditing the cask has no way of knowing if the package is what it claims to be or not**. This policy is very well documented in the main repo.

Seems like the homepage has no link to these fonts on their own, so closing. Thank you anyway for the PR.

**Comment by mhavo** [comment]
So I found this repo. Seems to be the official. How do you add fonts from that?

https://github.com/ArtifexSoftware/urw-base35-fonts

**Comment by commitay** [comment]
Take a look at: 

[font-alegreya](https://github.com/caskroom/homebrew-fonts/blob/47efb6bf64154cd0a027d35e47509e68673adf23/Casks/font-alegreya.rb)

https://github.com/caskroom/homebrew-fonts/blob/master/CONTRIBUTING.md

The upstream repo has OTF fonts, they are preferred over TTF.

**Comment by commitay** [comment]
https://github.com/caskroom/homebrew-fonts/pull/1404

**Pull request closed without merging**